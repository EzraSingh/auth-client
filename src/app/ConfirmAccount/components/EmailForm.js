import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const EmailForm = ({ handleSubmit }) => {
    return(
        <form onSubmit={handleSubmit}>
            <header>
                <h3>Account Recovery</h3>
                <p>Resend recovery link via email</p>
            </header>

            <label htmlFor="email">Email</label>
            <input id="email" name="email" type="email" required/>

            <input className="btn" type="submit" value="Send Recovery Link"/>
            <div className="prompt">
                <Link className="link" to="/register">Go back to registration</Link>
            </div>
        </form>
    )
}

EmailForm.propTypes = {
    handleSubmit: PropTypes.func.isRequired
}

export default EmailForm