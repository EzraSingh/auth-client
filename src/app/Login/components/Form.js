import React, { Component } from 'react'
import { Redirect, Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { login } from 'actions/auth'

class LoginForm extends Component{
    state = { reveal: false }
    handleSubmit = event => {
        event.preventDefault()
        const data = new FormData(event.target)
        this.props.login({ 
            email : data.get('email'),
            password : data.get('password')
        })
    }
    toggleVisibility = () => {
        this.setState({ reveal: !this.state.reveal })
    }
    render(){
        if(this.props.isLoggedIn){
            return <Redirect to='/profile'/>
        }
        return(
            <form onSubmit={this.handleSubmit}>
                <header>
                    <h3>Login</h3>
                    <p>Authenticate identity</p>
                </header>
                <label htmlFor="email">Email</label>
                <input id="email" name="email" type="email" required/>

                <label htmlFor="password">Password</label>
                <input id="password" name="password" type={this.state.reveal ? "text" : "password"} required/>
                <div className="hide-show">
                    <span onClick={this.toggleVisibility}>
                        {this.state.reveal ? 'Hide' : 'Show'}
                    </span>
                </div>
                <input className="btn" type="submit" value="Login"/>
                <div className="prompt">
                    <Link to='/register'>Don't have an account?</Link>
                </div>
            </form>
        )
    }
}

const mapStateToProps = state => {
    return { ...state.auth }
}

const mapDispatchToProps = dispatch => {
    return { login: credentials => dispatch(login(credentials)) }
}

export { LoginForm }
export default connect(mapStateToProps, mapDispatchToProps)(LoginForm)