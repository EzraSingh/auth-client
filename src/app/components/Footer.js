import React from 'react'

const Footer = () => {
    return(
        <footer>
            By 
            <a href="https://github.com/EzraSingh" target="_BLANK" rel="noopener noreferrer">
                Ezra Singh,
            </a>
            browse the source on
            <a href="https://github.com/EzraSingh/auth-client" target="_BLANK" rel="noopener noreferrer">
                <i className="fab fa-github"/>
            </a>
        </footer>
    )
}

export default Footer