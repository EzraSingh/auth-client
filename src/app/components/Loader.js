import React from 'react'

const Loader = () => {
    return <div className='sk-rotating-plane'/>
}

export default Loader