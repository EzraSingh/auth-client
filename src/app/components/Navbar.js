import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux'

/** Ruute template for registering routes within NavBar  */
function Route(name, href, auth, hideOnAuth){
    this.name = name
    this.href = href
    this.auth = auth
    this.hideOnAuth = hideOnAuth
}

class Navbar extends Component{
    static propTypes = {
        isLoggedIn: PropTypes.bool.isRequired,
        avatar: PropTypes.string,
        name: PropTypes.string
    }
    static routes = [
        new Route('Logout', '/logout', true)
    ]
    /** Filter routes that require authentication */
    generateRoutes(){
        if(!this.props.isLoggedIn){
            // Filter routes that do not require auth
            return Navbar.routes.filter((route) => !route.auth)
        }
        else{
            // Filter routes that require auth and are not flagged to hide on auth
            return Navbar.routes.filter((route) => route.auth && !route.hideOnAuth)
        }
    }
    renderAvatar = () =>{
        return(
            <figure className="avatar animated rotateIn delay-2">
                <NavLink to="/profile">
                    <img 
                        src={this.props.avatar || process.env.REACT_APP_DEFAULT_AVATAR} 
                        alt={this.props.name}
                    />
                </NavLink>
            </figure>
        )
    }
    render(){
        return(
            <nav className="flex navbar animated slideInDown">
                {this.generateRoutes().map((route) => (
                    <NavLink className="button button-outline" to={route.href} key={route.href}>
                        {route.name}
                    </NavLink>
                ))}
                {this.props.isLoggedIn && this.renderAvatar()}
            </nav>
        )
    }
}

const mapStateToProps = state => {
    return { ...state.auth, avatar: state.user.profile.avatar, name: state.user.profile.name }
}

export { Navbar as NavbarComponent }
export default connect(mapStateToProps)(Navbar)