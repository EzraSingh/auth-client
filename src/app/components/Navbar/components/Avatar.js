import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'

const Avatar = ({ avatar, name }) => {
    return(
        <NavLink id="avatar" to="/profile">
            <figure className="is-64 is-round">
                <img
                    src={avatar || process.env.REACT_APP_DEFAULT_AVATAR} 
                    alt={name}
                />
            </figure>
        </NavLink>
    )
}

Avatar.propTypes = {
    avatar: PropTypes.string,
    name: PropTypes.string.isRequired
}

export default Avatar