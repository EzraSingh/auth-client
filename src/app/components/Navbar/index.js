import React, { Fragment } from 'react'
import { default as Navbar } from './containers/MenuContainer'

const withNav = Component => props => {
    return(
        <Fragment>
            <Navbar/>
            <div className="navbar-container">
                <Component {...props}/>
            </div>
        </Fragment>
    )
}

export default Navbar
export { withNav }