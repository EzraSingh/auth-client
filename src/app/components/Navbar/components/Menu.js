import React, { Component } from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import { NavLink } from 'react-router-dom'
import Avatar from './Avatar'

class Menu extends Component{
    state = { showMenu: false }
    toggleMenu(){
        this.setState({ showMenu: !this.state.showMenu })
    }
    generateRoutes(){
        return this.props.routes.map((route) => (
            <NavLink className="link" to={route.href} key={route.href}>
                {route.name}
            </NavLink>
        ))
    }
    renderMenu(){
        return(
            <div className="container">
                {this.generateRoutes()}
                {this.props.isLoggedIn && <Avatar {...this.props.profile}/>}
            </div>
        )
    }
    renderMobileMenu(){
        const { showMenu } = this.state
        const toggleIcon = cx("fa", { "fa-window-close" : showMenu, "fa-bars" : !showMenu })
        return(
            <div id="mobile" className="container">
                <i 
                    onClick={this.toggleMenu.bind(this)}
                    className={toggleIcon}
                />
                <div className="menu">
                    {showMenu && this.generateRoutes()}
                </div>
            </div>
        )
    }
    render(){
        /** Toggles z-index to slide behind content when showMenu is false and overlay menu is active **/
        const smartHideStyle = { zIndex: this.state.showMenu ? 100 : 0 } 
        return(
            <nav style={smartHideStyle} className="navbar">
                {this.renderMenu()}
                {this.renderMobileMenu()}
            </nav>
        )
    }
}

Menu.propTypes = {
    profile: PropTypes.object.isRequired,
    routes: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        href: PropTypes.string.isRequired
    })).isRequired,
    isLoggedIn: PropTypes.bool.isRequired
}

export default Menu