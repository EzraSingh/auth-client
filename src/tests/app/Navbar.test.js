import React from 'react'
import { NavbarComponent } from 'app/components/Navbar'
import { shallow } from 'enzyme'

describe('<Navbar/>', () => {
    it('should render components w/o auth', () => {
        const props = { isLoggedIn: false, avatar: '', name: '' }
        const component = shallow(<NavbarComponent {...props}/>)
        expect(component).toMatchSnapshot()
    })
    it('should render components w/ auth', () => {
        const props = { isLoggedIn: true, avatar: '', name: '' }
        const component = shallow(<NavbarComponent {...props}/>)
        expect(component).toMatchSnapshot()
    })
})