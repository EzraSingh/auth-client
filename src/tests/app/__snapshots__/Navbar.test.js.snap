// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> should render components w/ auth 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    avatar=""
    isLoggedIn={true}
    name=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="button button-outline"
            to="/logout"
          >
            Logout
          </NavLink>,
        ],
        <figure
          className="avatar animated rotateIn delay-2"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/profile"
          >
            <img
              alt=""
              src="https://pngimage.net/wp-content/uploads/2018/05/default-user-profile-image-png-6.png"
            />
          </NavLink>
        </figure>,
      ],
      "className": "flex navbar animated slideInDown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "/logout",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Logout",
          "className": "button button-outline",
          "to": "/logout",
        },
        "ref": null,
        "rendered": "Logout",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            to="/profile"
          >
            <img
              alt=""
              src="https://pngimage.net/wp-content/uploads/2018/05/default-user-profile-image-png-6.png"
            />
          </NavLink>,
          "className": "avatar animated rotateIn delay-2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <img
              alt=""
              src="https://pngimage.net/wp-content/uploads/2018/05/default-user-profile-image-png-6.png"
            />,
            "to": "/profile",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "https://pngimage.net/wp-content/uploads/2018/05/default-user-profile-image-png-6.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        "type": "figure",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="button button-outline"
              to="/logout"
            >
              Logout
            </NavLink>,
          ],
          <figure
            className="avatar animated rotateIn delay-2"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/profile"
            >
              <img
                alt=""
                src="https://pngimage.net/wp-content/uploads/2018/05/default-user-profile-image-png-6.png"
              />
            </NavLink>
          </figure>,
        ],
        "className": "flex navbar animated slideInDown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "/logout",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Logout",
            "className": "button button-outline",
            "to": "/logout",
          },
          "ref": null,
          "rendered": "Logout",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/profile"
            >
              <img
                alt=""
                src="https://pngimage.net/wp-content/uploads/2018/05/default-user-profile-image-png-6.png"
              />
            </NavLink>,
            "className": "avatar animated rotateIn delay-2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <img
                alt=""
                src="https://pngimage.net/wp-content/uploads/2018/05/default-user-profile-image-png-6.png"
              />,
              "to": "/profile",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "https://pngimage.net/wp-content/uploads/2018/05/default-user-profile-image-png-6.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          "type": "figure",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Navbar/> should render components w/o auth 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    avatar=""
    isLoggedIn={false}
    name=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        false,
      ],
      "className": "flex navbar animated slideInDown",
    },
    "ref": null,
    "rendered": Array [
      false,
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          false,
        ],
        "className": "flex navbar animated slideInDown",
      },
      "ref": null,
      "rendered": Array [
        false,
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
